PWGGA/GammaConv/AliConversionMesonCuts.cxx

```diff
@@ -527,8 +527,8 @@ Bool_t AliConversionMesonCuts::MesonIsSelectedMCAODESD(AliDalitzAODESDMC *fMCMot
         // Select only -> 2y decay channel
         if(fMCMother->GetNDaughtersG()!=2)return kFALSE;
         if((fMCMother->GetLastDaughterG()<0) || (fMCMother->GetFirstDaughterG()<0)) return kFALSE;
-            AliDalitzAODESDMC *MDaughter0=mcEvent->Particle(fMCMother->GetFirstDaughterG());
-            AliDalitzAODESDMC *MDaughter1=mcEvent->Particle(fMCMother->GetLastDaughterG());
+            auto_ptr<AliDalitzAODESDMC> MDaughter0=mcEvent->Particle(fMCMother->GetFirstDaughterG());
```

Comment:
One more thing, it seems that some of these are kind-of overkill; in this case we're create a whole "new" particle to just get the PDGCode and then delete the particle immediately afterwords. It seems like it would be much more efficient to add a method to `AliDalitzEventMC`, something like:
```cpp
Int_t GetPgdCodeOfParticle(Int_t i)
{
  if (fIsESDMC) {
    return fESDEventMC->Particle(i)->GetPdgCode();
  } else {
    auto branch = AliAODMCParticle::StdBranchName();
    auto *array = static_cast<TClonesArray*>(fAODEvent->FindListObject(branch));
    auto *particle = static_cast<AliAODMCParticle*>(array->At(i));
    return particle->GetPDGCode();
  }
}
```

No `new`, no `auto_ptr`, no copying, just a lookup.

Then, these lines become:

```cpp
if (mcEvent->GetPgdCodeOfParticle(fMCMother->GetFirstDaughterG()) != 22 ||
    mcEvent->GetPgdCodeOfParticle(fMCMother->GetLastDaughterG()) != 22) {
   return kFALSE;
}
return kTRUE;
```

or more succinctly:

```cpp
// returns true if both pdg-codes are 22
return mcEvent->GetPgdCodeOfParticle(fMCMother->GetFirstDaughterG()) == 22 &&
       mcEvent->GetPgdCodeOfParticle(fMCMother->GetLastDaughterG()) == 22;
```

I'm PWGCF, not PWGGA, so I'll stop bothering you, just some suggestions.

Reply:
Dear Andrew,
Thanks a lot for the feedback, I really enjoy learning and see things I didn't see on the beginning, this last part of the call of the Daughter I use that way because I keep the same idea from the principle to create a switch class between AOD or ESD. So I create this class AliDalitzAODESDMC to get the info, the problem you just point me, I use for everything when some times is just better to make this lockout. I will see how all this work on the code and if is work -> make a commit 
Regards Edgar
(Copy and paste code sometimes never work, most of all the times... ;>)